diff --git a/libqalculate/util.cc b/libqalculate/util.cc
index 23b6da8..5d97581 100644
--- a/libqalculate/util.cc
+++ b/libqalculate/util.cc
@@ -556,91 +556,13 @@ string getOldLocalDir() {
 #endif
 }
 string getLocalDir() {
-	const char *homedir;
-	if((homedir = getenv("QALCULATE_USER_DIR")) != NULL) {
-		return homedir;
-	}
-#ifdef _WIN32
-	char exepath[MAX_PATH];
-	GetModuleFileName(NULL, exepath, MAX_PATH);
-	string str(exepath);
-	str.resize(str.find_last_of('\\'));
-	str += "\\user";
-#	ifdef WIN_PORTABLE
-	_mkdir(str.c_str());
-	return str;
-#	else
-	if(dirExists(str)) return str;
-	char path[MAX_PATH];
-	SHGetFolderPathA(NULL, CSIDL_LOCAL_APPDATA | CSIDL_FLAG_CREATE, NULL, 0, path);
-	str = path;
-	return str + "\\Qalculate";
-#	endif
-#else
-	if((homedir = getenv("XDG_CONFIG_HOME")) == NULL) {
-		return getHomeDir() + "/.config/qalculate";
-	}
-	return string(homedir) + "/qalculate";
-#endif
+	return string(getenv("SNAP_USER_DATA"));
 }
 string getLocalDataDir() {
-	const char *homedir;
-	if((homedir = getenv("QALCULATE_USER_DIR")) != NULL) {
-		return homedir;
-	}
-#ifdef _WIN32
-	char exepath[MAX_PATH];
-	GetModuleFileName(NULL, exepath, MAX_PATH);
-	string str(exepath);
-	str.resize(str.find_last_of('\\'));
-	str += "\\user";
-#	ifdef WIN_PORTABLE
-	_mkdir(str.c_str());
-	return str;
-#	else
-	if(dirExists(str)) return str;
-	char path[MAX_PATH];
-	SHGetFolderPathA(NULL, CSIDL_LOCAL_APPDATA | CSIDL_FLAG_CREATE, NULL, 0, path);
-	str = path;
-	return str + "\\Qalculate";
-#	endif
-#else
-	if((homedir = getenv("XDG_DATA_HOME")) == NULL) {
-		return getHomeDir() + "/.local/share/qalculate";
-	}
-	return string(homedir) + "/qalculate";
-#endif
+	return string(getenv("SNAP_USER_COMMON"));
 }
 string getLocalTmpDir() {
-	const char *homedir;
-	if((homedir = getenv("QALCULATE_USER_DIR")) != NULL) {
-		return homedir;
-	}
-#ifdef _WIN32
-	char exepath[MAX_PATH];
-	GetModuleFileName(NULL, exepath, MAX_PATH);
-	string str(exepath);
-	str.resize(str.find_last_of('\\'));
-#	ifdef WIN_PORTABLE
-	str += "\\tmp";
-	_mkdir(str.c_str());
-	return str;
-#	else
-	str += "\\user";
-	if(dirExists(str)) return str;
-	char path[MAX_PATH];
-	SHGetFolderPathA(NULL, CSIDL_LOCAL_APPDATA | CSIDL_FLAG_CREATE, NULL, 0, path);
-	str = path;
-	str += "\\cache";
-	_mkdir(str.c_str());
-	return str + "\\Qalculate";
-#	endif
-#else
-	if((homedir = getenv("XDG_CACHE_HOME")) == NULL) {
-		return getHomeDir() + "/.cache/qalculate";
-	}
-	return string(homedir) + "/qalculate";
-#endif
+	return string(getenv("SNAP_USER_DATA"));
 }
 
 bool move_file(const char *from_file, const char *to_file) {
@@ -681,7 +603,10 @@ bool move_file(const char *from_file, const char *to_file) {
 
 string getPackageDataDir() {
 #ifndef WIN32
-	return PACKAGE_DATA_DIR;
+	const char *snap = getenv("SNAP");
+	string datadir(snap);
+	if(dirExists(datadir + "/usr/share/qalculate")) {datadir += "/usr/share";} else {datadir += "/share";}
+	return datadir;
 #else
 	char exepath[MAX_PATH];
 	GetModuleFileName(NULL, exepath, MAX_PATH);
@@ -703,15 +628,7 @@ string getGlobalDefinitionsDir() {
 	return "";
 #else
 #	ifndef WIN32
-	char buffer[500];
-	if(getcwd(buffer, 500)) {
-		string dir = buffer;
-		if(dirExists(buildPath(dir, "libqalculate")) && fileExists(buildPath(dir, "data", "functions.xml"))) return buildPath(dir, "data");
-		size_t i = dir.rfind("/");
-		if(i != string::npos && i > 0 && i < dir.length() - 1) dir = dir.substr(0, i);
-		if(dirExists(buildPath(dir, "libqalculate")) && fileExists(buildPath(dir, "data", "functions.xml"))) return buildPath(dir, "data");
-	}
-	return string(PACKAGE_DATA_DIR) + "/qalculate";
+	return getPackageDataDir() + "/qalculate";
 #	else
 	char exepath[MAX_PATH];
 	GetModuleFileName(NULL, exepath, MAX_PATH);
@@ -742,7 +659,7 @@ string getGlobalDefinitionsDir() {
 
 string getPackageLocaleDir() {
 #ifndef WIN32
-	return PACKAGE_LOCALE_DIR;
+	return getPackageDataDir() + "/locale";
 #else
 	char exepath[MAX_PATH];
 	GetModuleFileName(NULL, exepath, MAX_PATH);
