diff --git a/data/io.github.Qalculate.qalculate-qt.desktop b/data/io.github.Qalculate.qalculate-qt.desktop
index 7504726..9816f30 100644
--- a/data/io.github.Qalculate.qalculate-qt.desktop
+++ b/data/io.github.Qalculate.qalculate-qt.desktop
@@ -28,7 +28,7 @@ Comment[zh_CN]=强大而易用的计算器
 Comment[zh_TW]=強大且易用的計算機
 TryExec=qalculate-qt
 Exec=qalculate-qt
-Icon=qalculate-qt
+Icon=usr/share/icons/hicolor/scalable/apps/qalculate-qt.svg
 Terminal=false
 Type=Application
 StartupNotify=true
